lib/embroider/update-manifest-from-html.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const fs = require('fs');
+const findScriptSrcs = require('find-scripts-srcs-in-document');
+const getScriptFileName = require('../utilities/get-script-file-name');
+
+module.exports = function updateManifestFromHtml(root) {
+  const pkg = JSON.parse(fs.readFileSync(root + '/package.json', 'UTF8'));
+
+  // otherwise, we must parse the index.html file to figure out what's next
+  const indexHtml = fs.readFileSync(root + '/index.html', 'UTF8')
+  // TODO: fix simple-html-tokenizer: https://github.com/tildeio/simple-html-tokenizer/pull/71
+    .replace('<!DOCTYPE html>','');
+
+  const assetsDir = `assets/`;
+  const vendorFiles = pkg.fastboot.manifest.vendorFiles;
+
+  // Find script src which are not data-fastboot-ignore
+  const indexScriptFilePaths = findScriptSrcs(indexHtml, findScriptSrcs.ignoreWithAttribute('data-fastboot-ignore'))
+  .filter( src => {
+    // skipping files if they are part of vendorFiles list
+    // this would cover any vendor files (vendor or vendor-static)
+    const fileName = getScriptFileName(src);
+    return !vendorFiles.find(filePath => fileName === getScriptFileName(filePath));
+  });
+
+  /**
+   * Reading app files by parsing index.html so that FastBoot can load assets that
+   * are same as what is generategd as part of the build flow and updated in
+   * index.html.
+   *
+   * Example: Embroider build flow upon WebPack as stage 3 processing in the build
+   * updates app.js to multiple chunks and updates index.html. Parsing from index.html
+   * allows loading same set of app files in FastBoot sandbox.
+   *
+   */
+  const appFiles = [];
+  indexScriptFilePaths.forEach(src => {
+    // Extract the file name from the src file paths and
+    // checks the file exits in the assets location on the disk.
+    const fileName = getScriptFileName(src);
+    const assetPath = `${assetsDir}${fileName}`;
+    const filePath = `${root}/${assetPath}`;
+    if(fileName && !appFiles.includes(assetPath) && fs.existsSync(filePath)) {
+      appFiles.push(assetPath);
+    }
```

Comment:
Wouldn't this include other script tags that may not be app files? Example I may add a google analytics script tag in my index.html. Seems like this might also be included when it shouldn't because we aren't checking for `data-fastboot-ignore` here. 

Also how would this work for LI apps that will have filename wrapped in spark placeholders?

Reply:
For the second question, I see you have added tests already. So I think that answers my second question.